packages/composite-checkout/src/lib/payment-methods/sofort.js

```diff
@@ -0,0 +1,241 @@
+import debugFactory from 'debug';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@automattic/react-i18n';
+
+/**
+ * Internal dependencies
+ */
+import Field from '../../components/field';
+import Button from '../../components/button';
+import {
+	usePaymentProcessor,
+	useTransactionStatus,
+	useLineItems,
+	renderDisplayValueMarkdown,
+	useEvents,
+} from '../../public-api';
+import { SummaryLine, SummaryDetails } from '../styled-components/summary-details';
+import { useFormStatus } from '../form-status';
+import { registerStore, useSelect, useDispatch } from '../../lib/registry';
+import { PaymentMethodLogos } from '../styled-components/payment-method-logos';
+
+const debug = debugFactory( 'composite-checkout:sofort-payment-method' );
+
+export function createSofortPaymentMethodStore() {
+	debug( 'creating a new sofort payment method store' );
+	const actions = {
+		changeCustomerName( payload ) {
+			return { type: 'CUSTOMER_NAME_SET', payload };
+		},
+	};
+
+	const selectors = {
+		getCustomerName( state ) {
+			return state.customerName || '';
+		},
+	};
+
+	const store = registerStore( 'sofort', {
+		reducer(
+			state = {
+				customerName: { value: '', isTouched: false },
+			},
+			action
+		) {
+			switch ( action.type ) {
+				case 'CUSTOMER_NAME_SET':
+					return { ...state, customerName: { value: action.payload, isTouched: true } };
+			}
+			return state;
+		},
+		actions,
+		selectors,
+	} );
+
+	return { ...store, actions, selectors };
+}
+
+export function createSofortMethod( { store, stripe, stripeConfiguration } ) {
+	return {
+		id: 'sofort',
+		label: <SofortLabel />,
+		activeContent: <SofortFields stripe={ stripe } stripeConfiguration={ stripeConfiguration } />,
+		submitButton: (
+			<SofortPayButton
+				store={ store }
+				stripe={ stripe }
+				stripeConfiguration={ stripeConfiguration }
+			/>
+		),
+		inactiveContent: <SofortSummary />,
+		getAriaLabel: ( __ ) => __( 'Sofort' ),
+	};
+}
+
+function SofortFields() {
+	const { __ } = useI18n();
+
+	const customerName = useSelect( ( select ) => select( 'sofort' ).getCustomerName() );
+	const { changeCustomerName } = useDispatch( 'sofort' );
+	const { formStatus } = useFormStatus();
+	const isDisabled = formStatus !== 'ready';
+
+	return (
+		<SofortFormWrapper>
+			<SofortField
+				id="cardholderName"
+				type="Text"
+				autoComplete="cc-name"
+				label={ __( 'Your name' ) }
+				value={ customerName?.value ?? '' }
+				onChange={ changeCustomerName }
+				isError={ customerName?.isTouched && customerName?.value.length === 0 }
+				errorMessage={ __( 'This field is required' ) }
+				disabled={ isDisabled }
+			/>
+		</SofortFormWrapper>
+	);
+}
+
+const SofortFormWrapper = styled.div`
+	padding: 16px;
+	position: relative;
+
+	:after {
+		display: block;
+		width: calc( 100% - 6px );
+		height: 1px;
+		content: '';
+		background: ${ ( props ) => props.theme.colors.borderColorLight };
+		position: absolute;
+		top: 0;
+		left: 3px;
```

Comment:
Need to handle `.rtl` here... maybe handling that manually isn't a good idea

Reply:
😬  We'll probably need to handle that for the other payment methods too. Maybe worth doing all at once?